packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-case-custom-pwd-recovery-mfa/main/java/siw.md

```diff
@@ -0,0 +1,124 @@
+### 1: Start password recovery
+
+To begin the password recovery flow, the user must:
+
+1. Click the **Forgot Password?** link on the sign-in page.
+2. Enter their **Email or Username** in the box and click **Next**.
+3. Choose **Email** as the authenticator they want to use for password recovery and click **Submit**.
+
+Okta then sends the user an email that matches the Forgot Password template that you altered earlier, and the Sign-In Widget tells them to click the link in the email or to enter the OTP to continue.
+
+<div class="common-image-format">
+
+![Screenshot of email sent to user](/img/advanced-use-cases/custom-pwd-recovery-custom-email.png "Password Recovery Email")
+
+</div>
+
+The email's **Reset Password** link includes the `otp` and `request.relayState` variables sent back as query parameters to the application. For example,
+
+`http://localhost:8080/magic-link/callback?otp=${oneTimePassword}&state=${request.relayState}` becomes `http://localhost:8080/magic-link/callback?otp=726009&state=1b34371af02dd31d2bc4c48a3607cd32`.
+
+### 2: Handle the OTP and state parameters
+
+Create a callback handler method that takes the `otp` and `state` parameters in the query string and passes them as session parameters to a page that contains the Sign-In Widget. First, retrieve the current `IDXClientContext` object from the browser session and check that `state` has a value.
+
+If `state` is `null` or the browser context object is `null` (because the user is clicking the magic link in a different browser), the widget will advise the user to return to the original tab in the browser where they requested a password reset and enter the OTP to proceed.
```

Comment:
```suggestion
If `state` is `null` or the browser context object is `null` (because the user clicked the magic link from a different browser), the widget instructs the user to return to the original tab in the browser where they requested a password reset and enter the OTP to proceed.
```

Reply:
fixed